x/pairing/README.md

```diff
@@ -0,0 +1,402 @@
+
+#### Stake
+
+When a provider stakes, a new stake entry is created on-chain. A stake entry is defined as follows:
+
+```go
+type StakeEntry struct {
+	Stake               Coin        // stake amount
+	Address             string      // provider Lava address
+	StakeAppliedBlock   uint64      // the block in which the stake is active
+	Endpoints           []Endpoint  // a list of endpoints
+	Geolocation         int32       // supported geolocation
+	Chain               string      // chain in which the provider is staked on
+	Moniker             string      // non-unique human name
+	DelegateTotal       Coin        // total delegations
+	DelegateLimit       Coin        // max amount of delegations the provider accepts
+	DelegateCommission  uint64      // commission for delegation
+}
+```
+
+Note, the `Coin` type is from Cosmos-SDK (`cosmos.base.v1beta1.Coin`). A provider can accept delegations to increase its effective stake, which increases its chances of being selected in the pairing process. The provider can also set a delegation limit, which determines the maximum value of delegations they can accept. This limit is in place to prevent delegators from increasing the provider's effective stake to a level where the provider is overwhelmed with more consumers than they can handle in the pairing process. For more details about delegations, refer to the dualstaking module README.
+
+An provider's endpoint is defined as follows:
+
+```go
+type Endpoint struct {
+	IPPORT        string    // IP + port
+	Geolocation   int32     // supported geolocation
+	Addons        []string  // list of supported add-ons
+	ApiInterfaces []string  // list of supported API interfaces
+	Extensions    []string  // list of supported extensions
+}
+```
+
+A consumer sends requests to a provider's endpoint to communicate with them. The aggregated geolocations of all the provider's endpoints should be equal to the geolocation field present in the provider's stake entry. Other than that, the `Addons`, `ApiInterfaces` and `Extensions` specify the "extra" features the provider supports. To get more details about those, see the spec module README.
+
+Stake entries' storage is managed by the epochstorage module. For more details, see its README.
+
+#### Unstake
+
+A provider can unstake and retrieve their coins. When a provider unstakes, they are removed from the pairing list starting from the next epoch. After a specified number of blocks called `UnstakeHoldBlocks` (a parameter of the epochstorage module), the provider is eligible to receive their coins back.
+
+#### Freeze
+
+Freeze Mode enables the Provider to temporarily suspend their node's operation during maintenance to avoid bad Quality of Service (QoS). Freeze/Unfreeze is applied on the next Epoch. The Provider can initiate multiple Freeze actions, but only one Freeze-per-chain can be take effect at a time.
```

Comment:
remove 
"but only one Freeze-per-chain can be take effect at a time"

Reply:
done. see https://github.com/lavanet/lava/commit/6de4aabf932a0ccf06e72626912782c8de252add